apps/src/codebridge/hooks/useSource.ts

```diff
@@ -30,6 +30,7 @@ export const useSource = (defaultSources: ProjectSources) => {
   );
   const previousChannelIdRef = useRef<string | null>(null);
   const levelId = useAppSelector(state => state.lab.levelProperties?.id);
+  const [hasSetInitialSource, setHasSetInitialSource] = useState(false);
```

Comment:
No suggestion, but `setHasSetInitialSource` is hard for my brain to parse.

Reply:
I ended up reworking this so I didn't need this check anymore. Instead of gating setting initial sources on a channel id change, we now check based on level id changing. Both start and exemplar mode don't have channel ids, but they have level ids, and it's not possible for the channel to change but not the level.